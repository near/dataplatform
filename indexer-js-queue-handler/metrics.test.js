import { jest } from '@jest/globals';

import Metrics from './metrics';

describe('Metrics', () => {
    const oldEnv = process.env;

    beforeAll(() => {
        process.env = {
            ...oldEnv,
            STAGE: 'dev',
        };
    });

    afterAll(() => {
        process.env = oldEnv;
    });

    it('writes the block height for an indexer function', async () => {
        const cloudwatch = {
            putMetricData: jest.fn().mockReturnValueOnce({ promise: jest.fn() })
        };
        const metrics = new Metrics('test', cloudwatch);

        await metrics.putBlockHeight('morgs.near', 'test', false, 2);

        expect(cloudwatch.putMetricData).toBeCalledTimes(1);
        expect(cloudwatch.putMetricData.mock.calls[0]).toMatchSnapshot()
    });
})
