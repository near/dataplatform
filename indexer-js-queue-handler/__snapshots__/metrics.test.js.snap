// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics writes the block height for an indexer function 1`] = `
[
  {
    "MetricData": [
      {
        "Dimensions": [
          {
            "Name": "ACCOUNT_ID",
            "Value": "morgs.near",
          },
          {
            "Name": "FUNCTION_NAME",
            "Value": "test",
          },
          {
            "Name": "STAGE",
            "Value": "dev",
          },
        ],
        "MetricName": "INDEXER_FUNCTION_LATEST_BLOCK_HEIGHT",
        "Unit": "None",
        "Value": 2,
      },
    ],
    "Namespace": "test",
  },
]
`;
